packages/blocks/src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -0,0 +1,130 @@
+import { PlusIcon, SearchIcon } from '@mergestat/icons';
+import React, { useState } from 'react';
+import { Tag } from '../Filter/Tag';
+
+type MultiSelectProps = {
+  setStateToProps: {
+    title: string;
+    checked: boolean;
+  }[];
+  getState?: ([]) => void;
+};
+
+export const MultiSelect: React.FC<
+  MultiSelectProps &
+    React.DetailedHTMLProps<
+      React.DetailsHTMLAttributes<HTMLElement>,
+      HTMLElement
+    >
+> = ({ setStateToProps, getState }) => {
+  const [state, setState] = useState(setStateToProps);
+  const [value, setValue] = useState('');
+  const [isActive, setIsActive] = useState(false);
+
+  window.addEventListener('click', (e) => {
+    const target = e.target as Element
+    if (target.localName !== 'input' && target.localName !== 'label') {
+      setIsActive(false);
+    }
+  });
+
+  window.addEventListener('keypress', (e) => {
+    if (e.key === 'Enter' && value.length > 0) {
+      setState([...state, { title: value, checked: true }]);
+
+      if (getState) {
+        getState([...state]);
+      }
+      setValue('');
+    }
+  });
```

Comment:
it _might_ be worth checking out this library: https://react-select.com/home if we can customize it enough. See [here](https://react-select.com/components#replacing-components)